feat(docs): How-to guides about UP4W agent startup (#944)
canonical/ubuntu-pro-for-wsl: docs/howto/enforce-agent-startup-remotely-registry.md

4. The next time the user logs on the Windows device, the OS will execute the UP4W startup task (whether had the user
   interacted with the application or not).
5. Windows remote management solutions can monitor that registry key value and proactively fix it, thus enforcing the
   UP4W startup task to be always enabled.

(howto::enforce-with-intune)=
## Using Intune Remediations

Remediations are script packages that can detect and fix common issues on a user's device before they even realise
there's a problem. Each script package consists of a detection script, a remediation script, and metadata. Through
Intune, you can deploy these script packages and see reports on their effectiveness.
[Visit the Microsoft Intune documentation](https://learn.microsoft.com/en-us/mem/intune/fundamentals/remediations)
to learn more about it. Those scripts run on a predefined schedule and if the detection script reports a failure (by
`exit 1`) then the remediation script will run. That allows system administrators to watch for the specific Registry
key value that represents the enablement of the UP4W background agent startup task. The contents of both scripts are

> presented below. **Make sure to save them encoded in UTF-8**, as required by Intune.

Author: TBH I don't know how obvious would it be but I considered that out of scope of this guide. We'd be in the realm of documenting how to use text editors.

![Scripts and remediations option revealed in the Intune portal](./assets/intune-remediations.png).

Click on the "Create" button to create a new script package. Fill in the **Basics** step form with name, description and other details and proceed to **Settings**.
On that step upload the scripts in the correct boxes and finish the options on that step with:

- Run this script using the logged-on credentials (important because the script refers to a registry path under `HKCU`
  a.k.a `HKEY_CURRENT_USER`)
- Enforce script signature check: No (unless otherwise required by your company's policies)
- Run script in 64-bit PowerShell: No

Finally select "Next" and assign "Scope tags" (if used in your company and/or use case) and in the "Assignments" select
the device or user groups as required to reach the intended audience.

Follow [this guide](https://learn.microsoft.com/en-us/mem/intune/fundamentals/remediations#deploy-the-script-packages)
if you need more detailed views of the steps outlined above.

At next logon users covered by the assignment will have Intune executing the detection script and the remediation one
if the device is found non-compliant.

## Remarks

> Careful readers would notice that there is an inherent race condition between setting the registry value and installing
> the MSIX (if remotely deployed): when the MSIX is installed the referred registry sub-key is recreated, overwriting any
> previous value the remote management solution would have deployed if that happened before the package installation.

Reviewer: ```suggestion
Careful readers may notice that there is an inherent race condition between setting the registry value and installing
the MSIX (if remotely deployed): when the MSIX is installed the referred registry sub-key that is referenced is recreated, overwriting any
previous value that the remote management solution would have deployed if that happened before the package installation.
```

> One advantage of Intune Remediation scripts in this scenario is that eventually Intune would find the non-compliant
> state and fix it automatically. One disadvantage of that scenario is that the fix doesn't start the UP4W background
> agent, i.e. it only enables the startup task, the agent will start at next logon.

Reviewer: ```suggestion
An advantage of Remediation scripts in this scenario is that eventually Intune finds the non-compliant
state and fixes it automatically. A potential disadvantage is that the fix doesn't start the UP4W background
agent. The fix enables the startup task and the agent will then start at next user login.
```

Other material that might be candidate for explanation article.
